var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -21,12 +40,50 @@ class PyScikitLearn(PythonPackage):
     version('0.17.1', 'a2f8b877e6d99b1ed737144f5a478dfc')
     version('0.13.1', 'acba398e1d46274b8470f40d0926e6a4')
 
-    depends_on('python@2.6:2.8,3.3:', when='@:0.19.1')
-    depends_on('python@2.7:2.8,3.4:', when='@0.20.0:')
-    depends_on('py-numpy@1.6.1:', type=('build', 'run'), when='@:0.19.1')
-    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.20.0:')
-    depends_on('py-scipy@0.9:', type=('build', 'run'), when='@:0.19.1')
-    depends_on('py-scipy@0.13.3:', type=('build', 'run'), when='@0.20.0:')
+    depends_on('python@2.6:2.8,3.3:', when='@:0.19')
+    depends_on('python@2.7:2.8,3.4:', when='@0.20.0:0.20.999')
+    depends_on('python@3.5:', when='@0.21:')
+    depends_on('py-numpy@1.6.1:', type=('build', 'run'), when='@:0.19')
+    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.20.0:0.20.999')
+    depends_on('py-numpy@1.11.0:', type=('build', 'run'), when='@0.21:')
+    depends_on('py-scipy@0.9:', type=('build', 'run'), when='@:0.19')
+    depends_on('py-scipy@0.13.3:', type=('build', 'run'), when='@0.20.0:0.20.999')
+    depends_on('py-scipy@0.17.0:', type=('build', 'run'), when='@0.21:')
+    depends_on('py-joblib@0.11:', type=('build', 'run'))
```

Comment:
It looks like sklearn always depended on joblib, but prior to 0.21.0 it was a vendored dependency.